### PR TITLE
Recursive mutation

### PR DESCRIPTION
**Pull request by manaskarekar**
Fix issue referenced in #40 and #32.

### CODE REVIEW

**Comment by manaskarekar** [comment]
Sorry about the second PR.

**Comment by llogiq** [comment]
I had to merge another PR now, so we got a merge conflict. Can you resolve the conflict or do you need help with that? Sorry for the inconvenience.

**Comment by manaskarekar** [comment]
I've resolved the merge conflicts. Please let me know if something doesn't look right.

`cargo +nightly test` on plugins does give me 'mismatched types' error at the mutate attribute but I believe that is being addressed in another issue (I haven't pulled from master recently).

File: plugin/src/lib.rs [hunk 1]
```diff
@@ -319,9 +319,9 @@ impl<'a, 'cx> Folder for MutatorPlugin<'a, 'cx> {
                             ],
                         );
                     }
-                    let left = left.map(|e| fold::noop_fold_expr(e, self));
-                    let right = right.map(|e| fold::noop_fold_expr(e, self));
-                    quote_expr!(self.cx, ::mutagen::and(|| $left, || $right, $n))
```
Comment (hmvp) [inline plugin/src/lib.rs]: Am I correct when I say that you should not remove the `::` in the last line? I think that was actually added in another commit recently

File: plugin/tests/binops.rs [hunk 2]
```diff
@@ -89,7 +93,6 @@ fn test_binop_ne() {
     assert!(checker.has_multiple(noneq_msgs, "26:9: 26:15"));
 }
 
-#[test]
```
Comment (hmvp) [inline plugin/tests/binops.rs]: I fail to see why the is not a test anymore. Is this intentional?

Comment (manaskarekar) [reply plugin/tests/binops.rs]: Sorry, I messed up the merge.

**Comment by llogiq** [comment]
Yeah, merges are hard. Anyway, this looks good. Thank you for bearing with us! :+1:

**Pull request merged**